spec/ggt.spec.ts

```diff
@@ -46,4 +47,26 @@ describe("ggt", () => {
 
     await aborted;
   });
+
+  it("exits early if running in the Gadget editor's terminal", async () => {
+    mock(process, "env", "get", () => {
+      return {
+        GADGET_EDITOR_TERMINAL_SESSION_ID: "123",
+      };
+    });
+    let receivedExitCode: number | undefined;
+    mock(process, "exit", (code?: number): never => {
+      receivedExitCode = code;
+      throw new Error("This is a mock exit");
+    });
+
+    try {
+      await ggt(testCtx);
+    } catch (error) {
+      expect((error as any).message).toEqual("This is a mock exit");
+    }
+
+    expectStdout().toEqual("Running ggt in the Gadget editor's terminal is not supported.\n");
+    expect(receivedExitCode).toBe(1);
```

Comment:
nit: testing `process.exit` is pretty common in ggt so we have a helper named `expectProcessExit` that you can use! You can also use `vi.stubEnv` to mock environment variables.

```suggestion
    vi.stubEnv("GADGET_EDITOR_TERMINAL_SESSION_ID", "123");
    
    await expectProcessExit(() => ggt(testCtx), 1);

    expectStdout().toEqual("Running ggt in the Gadget editor's terminal is not supported.\n");
```